### PR TITLE
Add text highliter in dialog box

### PR DESCRIPTION
**Pull request by manojkrishna1991**

### CODE REVIEW

File: src/dialog/dialog.test.tsx [hunk 1]
```diff
@@ -1,6 +1,5 @@
-import * as React from 'react';
-
 import { cleanup, fireEvent, render } from '@config/testing';
+import * as React from 'react';
```
Comment (BenjaminEckardt) [inline src/dialog/dialog.test.tsx:2]: Do these changes have an intention?

File: src/dialog/dialog.tsx [hunk 2]
```diff
@@ -1,7 +1,6 @@
-import * as React from 'react';
-
 import { Button } from '@ampel-ui/button';
 import { onOuterClick } from '@ampel-ui/subscriptions';
+import * as React from 'react';
```
Comment (BenjaminEckardt) [inline src/dialog/dialog.tsx:3]: Do these changes have an intention?

File: src/text-highlighter/text-highlighter.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+class TextHighLighter extends React.Component<Props, {}> {
+    public render() {
+        const parts = this.props.content.split(new RegExp(`(${this.props.highlightedTexts.join('|')})`, 'g'));
+        return (
+            <div>
+                {parts.map((part, i) => (
+                    <span
+                        data-qa={`highlighted-${part}`}
+                        key={i}
+                        style={this.props.highlightedTexts.includes(part) ? { fontWeight: 'bold' } : {}}
```
Comment (BenjaminEckardt) [inline src/text-highlighter/text-highlighter.tsx:17]: Shouldn't that be configurable? What about adding a certain class, so the styling can be done outside?

File: src/text-highlighter/text-highlighter.test.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+import { TextHighLighter } from './text-highlighter';
+
+describe('Dialog', () => {
+    it('should render with highlightedText', () => {
+        const content = 'Message admin';
+        const { getByDataQa } = render(<TextHighLighter content={content} highlightedTexts={['admin']} />);
+        const highlightedText = getByDataQa('highlighted-admin');
+        expect(highlightedText).toBeTruthy();
+        expect(highlightedText.style.fontWeight).toBe('bold');
```
Comment (BenjaminEckardt) [inline src/text-highlighter/text-highlighter.test.tsx:12]: As mentioned in the implementation: hard coded style is not very flexible. Classes would also be easier to test

File: src/text-highlighter/text-highlighter.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import * as React from 'react';
+
+interface Props {
+    highlightedTexts: Array<string>;
+    content: string;
+}
+
+class TextHighLighter extends React.Component<Props, {}> {
+    public render() {
+        const parts = this.props.content.split(new RegExp(`(${this.props.highlightedTexts.join('|')})`, 'g'));
```
Comment (BenjaminEckardt) [inline src/text-highlighter/text-highlighter.tsx:10]: What happens when you pass a keyword containing `|`. Can you think of any other input that line might be sensitive to?

**Review by BenjaminEckardt — CHANGES_REQUESTED** [review]

File: src/dialog/dialog.test.tsx (hunk 1 above)
Comment (manojkrishna1991) [reply src/dialog/dialog.test.tsx:2]: auto-formatting i have reverted the changes

File: src/dialog/dialog.tsx (hunk 2 above)
Comment (manojkrishna1991) [reply src/dialog/dialog.tsx:3]: auto-formatting i have reverted the changes

File: src/text-highlighter/text-highlighter.tsx (hunk 3 above)
Comment (manojkrishna1991) [reply src/text-highlighter/text-highlighter.tsx]: fixed changed it to a class

File: src/text-highlighter/text-highlighter.test.tsx (hunk 4 above)
Comment (manojkrishna1991) [reply src/text-highlighter/text-highlighter.test.tsx]: fixed

File: src/text-highlighter/text-highlighter.tsx (hunk 5 above)
Comment (manojkrishna1991) [reply src/text-highlighter/text-highlighter.tsx:10]: checking on this it currently fails with | cases

Comment (manojkrishna1991) [reply src/text-highlighter/text-highlighter.tsx:10]: i have identified the possible list !@#$%^&*()+=\[\]\\';,./{}|":<>?~_-

Comment (manojkrishna1991) [reply src/text-highlighter/text-highlighter.tsx]: fixed

**Review by BenjaminEckardt — APPROVED** [review]

**Pull request merged**